bundle/terranova/tnresources/all_test.go

```diff
@@ -110,12 +110,47 @@ func testCRUD(t *testing.T, group string, adapter *Adapter, client *databricks.W
 
 	ctx := context.Background()
 
-	createdID, err := adapter.DoCreate(ctx, config)
+	// initial DoRefresh() cannot find the resource
+	remote, err := adapter.DoRefresh(ctx, "1234")
+	require.Nil(t, remote)
+	require.Error(t, err)
+	// TODO: if errors.Is(err, databricks.ErrResourceDoesNotExist) {... }
+
+	createdID, remoteStateFromCreate, err := adapter.DoCreate(ctx, config)
 	require.NoError(t, err, "DoCreate failed config=%v", config)
 	require.NotEmpty(t, createdID, "ID returned from DoCreate was empty")
 
-	err = adapter.DoUpdate(ctx, createdID, config)
+	remote, err = adapter.DoRefresh(ctx, createdID)
+	require.NoError(t, err)
+	require.NotNil(t, remote)
+	if remoteStateFromCreate != nil {
+		require.Equal(t, remoteStateFromCreate, remote)
+	}
+
+	remoteStateFromWaitCreate, err := adapter.WaitAfterCreate(ctx, config)
+	require.NoError(t, err)
+	if remoteStateFromWaitCreate != nil {
+		require.Equal(t, remote, remoteStateFromWaitCreate)
+	}
+
+	remoteStateFromUpdate, err := adapter.DoUpdate(ctx, createdID, config)
 	require.NoError(t, err, "DoUpdate failed")
+	if remoteStateFromUpdate != nil {
+		require.Equal(t, remote, remoteStateFromUpdate)
+	}
+
+	remoteStateFromWaitUpdate, err := adapter.WaitAfterUpdate(ctx, config)
+	require.NoError(t, err)
+	if remoteStateFromWaitCreate != nil {
```

Comment:
should be `remoteStateFromWaitUpdate`?

Reply:
good catch, ec88f42ae29b5cb050b0925940fd8a68f94b3ccd